### PR TITLE
Changed min/max workers to single field

### PR DESCRIPTION
**Pull request by Maxusmusti**
# Issue link
#232 

# What changes have been made
Replaced `min_worker` and `max_worker` with a single consistent worker field in cluster config

# Verification steps
Create a cluster
check `cluster.details()`
check the actual cluster details

## Checks
- [x] I've made sure the tests are passing. 
- Testing Strategy
   - [x] Unit tests
   - [ ] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

**Comment by Maxusmusti** [comment]
One important note, I've noticed a couple of other things while making these changes:
 - I named the config field `worker`, to be consistent with `gpu`. Should we update these to `workers` and `gpus` for clarity, to show we are specifying an amount? Or is it fine as is?
  - I noticed that `cluster.details()` and our internal `RayCluster` object only show/hold one value for `cpu`, even though we actually have a `min_cpu` and `max_cpu`. Should these be updated to also show/hold both the min and the max properly? (aka the opposite problem of what we had with this PR lol)

**Comment by tedhtchang** [comment]
It works try the following 
```
cluster = Cluster(ClusterConfiguration(local_interactive=local_interactive, namespace=namespace, name=cluster_name, worker=1, min_cpus=1, max_cpus=1, min_memory=4, max_memory=4, gpu=0, instascale=False, machine_types=["m5.xlarge", "p3.8xlarge"]))
```

image

**Comment by jbusche** [comment]
Looks good Mustafa, tried it with:  
```
cluster = Cluster(ClusterConfiguration(
    name='mnisttest',
    namespace='default',
    worker=2,
    min_cpus=2,
    max_cpus=2,
    min_memory=2,
    max_memory=2,
    gpu=0,
    image="quay.io/project-codeflare/ray:2.5.0-py38-cu116",
    instascale=False, # Can be set to false if scaling not needed
    machine_types=["m5.xlarge", "g4dn.xlarge"] # Can be removed if above is false
))
```

**Review by tedhtchang — COMMENTED** [review]
May need to update these notebooks
demo-notebooks/guided-demos/notebook-ex-outputs/0_basic_ray.ipynb
demo-notebooks/guided-demos/notebook-ex-outputs/3_basic_interactive.ipynb
demo-notebooks/guided-demos/notebook-ex-outputs/2_basic_jobs.ipynb
demo-notebooks/guided-demos/notebook-ex-outputs/4_gpt.ipynb

File: src/codeflare_sdk/utils/pretty_print.py [hunk 1]
```diff
@@ -142,10 +141,10 @@ def print_clusters(clusters: List[RayCluster]):
         #'table1' to display the worker counts
         table1 = Table(box=None)
         table1.add_row()
-        table1.add_column("Min", style="cyan", no_wrap=True)
-        table1.add_column("Max", style="magenta")
+        # table1.add_column("Min", style="cyan", no_wrap=True)
```
Comment (MichaelClifford) [inline src/codeflare_sdk/utils/pretty_print.py:144]: should we just delete this line?

**Review by MichaelClifford — CHANGES_REQUESTED** [review]

**Review by tedhtchang — COMMENTED** [review]
/lgtm keeping the old notebooks is a great idea.

**Comment by jbusche** [comment]
/LGTM Thanks Mustafa, I was able to run on my cluster.  Still not quite certain how it all gets folded in so the num_workers and num_gpus begin to work with the codeflare_notebook, but LGTM in testing.

**Review by jbusche — APPROVED** [review]
Tested OK on my OC 4.12 cluster

**Review by MichaelClifford — APPROVED** [review]
Thanks @Maxusmusti changes look good! :rocket:

**Pull request merged**